Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/VehicleRegistrationCertificate.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE ApplicativeDo #-}
+
+module SharedLogic.VehicleRegistrationCertificate
+  ( getAllLinkedRCs,
+    LinkedRCStatus (..),
+    idfyToLinkedRCStatus,
+    ResponseStatus (..),
+    verificationStatus,
+  )
+where
+
+import qualified Data.HashMap.Strict as HM
+import qualified Domain.Types.DriverOnboarding.IdfyVerification as IV
+import Domain.Types.DriverOnboarding.Image
+import qualified Domain.Types.Person as Person
+import Environment
+import Kernel.External.Encryption
+import Kernel.Prelude
+import Kernel.Types.Id
+import Storage.Queries.DriverOnboarding.DriverRCAssociation (buildRcHM)
+import qualified Storage.Queries.DriverOnboarding.DriverRCAssociation as DAQuery
+import qualified Storage.Queries.DriverOnboarding.Image as IQuery
+import qualified Storage.Queries.DriverOnboarding.VehicleRegistrationCertificate as RCQuery
+
+data LinkedRCStatus = LinkedRCStatus
+  { rcNumber :: Text,
+    isActive :: Bool,
+    status :: ResponseStatus
+  }
+  deriving (Show, Eq, Read, Generic, ToSchema, ToJSON, FromJSON)
+
+data ResponseStatus = NO_DOC_AVAILABLE | PENDING | VALID | FAILED | INVALID | LIMIT_EXCEED | MANUAL_VERIFICATION_REQUIRED
+  deriving (Show, Eq, Read, Generic, ToJSON, FromJSON, ToSchema, ToParamSchema, Enum, Bounded)
+
+getAllLinkedRCs :: Id Person.Person -> Flow [LinkedRCStatus]
+getAllLinkedRCs driverId = do
+  allLinkedRCs <- DAQuery.findAllLinkedByDriverId driverId
+  rcs <- RCQuery.findAllById (map (.rcId) allLinkedRCs)
+  let activeRcs = buildRcHM allLinkedRCs
+  mapM (getCombinedRcData activeRcs) rcs
+  where
+    getCombinedRcData activeRcs rc = do
+      rcNo <- decrypt rc.certificateNumber
+      return $
+        LinkedRCStatus
+          { rcNumber = rcNo,
+            isActive = Just True == (HM.lookup rc.id.getId activeRcs <&> (.isRcActive)),
+            status = VALID
+          }
+
+idfyToLinkedRCStatus :: Id Person.Person -> ImageType -> Int -> IV.IdfyVerification -> Flow LinkedRCStatus
+idfyToLinkedRCStatus driverId docType driverOnBoardingLimit verification = do
+  images <- IQuery.findRecentByPersonIdAndImageType driverId docType
+  rcNumber <- decrypt verification.documentNumber
+  let verificationStatus_ = verificationStatus driverOnBoardingLimit (length images) (Just verification)
+  return $
+    LinkedRCStatus
+      { rcNumber = rcNumber,
+        isActive = False,
+        status = verificationStatus_
+      }
+
+verificationStatus :: Int -> Int -> Maybe IV.IdfyVerification -> ResponseStatus
+verificationStatus onboardingTryLimit imagesNum verificationReq =
+  case verificationReq of
+    Just req -> do
+      if req.status == "completed"
```

Comment:
Not a subject of your PR. But can we make here more clear type instead of `Text`?